app/channels/new/controller.js

```diff
@@ -26,20 +26,31 @@ export default Ember.Controller.extend(Validations, {
 
 	// cleans the slug from bad things and suffix it with a random string
 	cleanSlug: computed('title', function () {
-		const title = clean(this.get('title'));
-		const random = randomText();
-		return `${title}-${random}`;
+		return clean(this.get('title'));
 	}),
 
+	suffixSlug(slug) {
+		const random = randomText();
+		return `${slug}-${random}`;
+	},
+
 	createRadio: task(function * (event) {
 		event.preventDefault();
 		const messages = get(this, 'flashMessages');
 		const user = get(this, 'session.currentUser');
-		const slug = get(this, 'cleanSlug');
-		let title = get(this, 'title');
+		const title = get(this, 'title').trim();
+		let slug = get(this, 'cleanSlug');
 
-		// Avoid extra spaces
-		title = title.trim();
+		// does a channel with this slug already exists
+		yield this.store.query('channel', {
+			orderBy: 'slug',
+			equalTo: slug
+		}).then(data => {
```

Comment:
since we're using `yield` there's no need to use `.then`.

```
const data = yield this.store.query....
const channelSlugExists = data.get('firstObject');
```